spec/activerecord-bitemporal/callbacks_spec.rb

```diff
@@ -0,0 +1,218 @@
+          it { expect { subject }.not_to change(employee, :log) }
+          it { expect { subject }.to change { employee.job_title.log }.from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+        end
+
+        context 'address with option `ignore_bitemporal_callbacks: true`' do
+          before { employee.job_title.bitemporal_option_merge!(ignore_bitemporal_callbacks: true) }
+          it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+          it { expect { subject }.not_to change { employee.job_title.log } }
+        end
+      end
+
+      context 'with _assign_attribute' do
+        subject {
+          employee.name = "Tom"
+          employee.job_title.name = "COO"
+          employee.save!
+        }
+
+        context 'without option `ignore_bitemporal_callbacks`' do
+          it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+          it { expect { subject }.to change { employee.job_title.log }.from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+        end
+
+        context 'employee with option `ignore_bitemporal_callbacks: true`' do
+          before { employee.bitemporal_option_merge!(ignore_bitemporal_callbacks: true) }
+          it { expect { subject }.not_to change(employee, :log) }
+          it { expect { subject }.to change { employee.job_title.log }.from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+        end
+
+        context 'address with option `ignore_bitemporal_callbacks: true`' do
+          before { employee.job_title.bitemporal_option_merge!(ignore_bitemporal_callbacks: true) }
+          it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_update after_bitemporal_update]) }
+          it { expect { subject }.not_to change { employee.job_title.log } }
+        end
+      end
+    end
+  end
+
+  describe "bitemporal_destroy" do
+    let(:employee) { EmployeeWithBitemporalCallbacksLog.create!(name: "Jane").tap(&:clear_log) }
+    subject { employee.destroy! }
+
+    context 'without option `ignore_bitemporal_callbacks`' do
+      it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_destroy after_bitemporal_destroy]) }
+    end
+
+    context 'with option `ignore_bitemporal_callbacks: false`' do
+      before { employee.bitemporal_option_merge!(ignore_bitemporal_callbacks: false) }
+      it { expect { subject }.to change(employee, :log).from([]).to(%i[before_bitemporal_destroy after_bitemporal_destroy]) }
+    end
+
+    context 'with option `ignore_bitemporal_callbacks: true`' do
+      before { employee.bitemporal_option_merge!(ignore_bitemporal_callbacks: true) }
+      it { expect { subject }.not_to change(employee, :log) }
+    end
+
+    context 'has_one with `dependent: :destroy`' do
```

Comment:
Do we also need a test for nested attributes (`{ job_title_attributes: { _destroy: true } }`)?

Reply:
I added nested attributes test case for bitemporal_destroy.